data/virt_autotest/clean_up_virt_logs.sh

```diff
@@ -0,0 +1,173 @@
+	   echo "YES"
+	else
+	   echo "NO"
+	fi
+}
+
+#Determine whether log file is empty file
+function is_empty_file() {
+	local log_file=$1
+
+	if [[ `file ${log_file} | grep ":.*empty"` ]];then
+	   echo "YES"
+	else
+	   echo "NO"
+	fi
+}
+
+#Empty text log file or rm empty text log file
+function cleanup_single_file() {
+	local file_path=$1
+
+	is_text=`is_text_file ${file_path}`
+	is_empty=`is_empty_file ${file_path}`
+	if [[ ${is_text} == "YES" ]];then
+	   > ${file_path}
+	   echo -e "Cleaned up text log file ${file_path}."
+	   return 1
+	elif [[ ${is_empty} == "YES" ]];then
+	   rm -f -r ${file_path}
+	   echo -e "Removed empty log file ${file_path}."
+	   return 1
+	else
+	   return 0
+	fi
+}
+
+#Clean log files or folders by using recursive algorithm
+function cleanup_logs() {
+	local logs_path=$1
+	local total_text_file=0
+	local is_text=""
+	local is_empty=""
+	local eachlog=""
+
+	if [[ -f ${logs_path} ]];then
+	   cleanup_single_file ${logs_path}
+	   total_text_file=$(( ${total_text_file} + $? ))
+	elif [[ -d ${logs_path} ]];then
+	   if [[ `echo ${logs_path} | tr '[A-Z]' '[a-z]'` =~ .*image.* ]];then
+	      echo -e "Skip image folder ${logs_path}."
+	   else
+	      for eachlog in `ls ${logs_path}`;do
+	          if [[ -f ${logs_path}/${eachlog} ]];then
+	             cleanup_single_file ${logs_path}/${eachlog}
+	             total_text_file=$(( ${total_text_file} + $? ))
+	          else
+	             pushd ${logs_path}/${eachlog} &> /dev/null
+	             cleanup_logs ${logs_path}/${eachlog}
+	             total_text_file=$(( ${total_text_file} + $? ))
+	             popd &> /dev/null
+	          fi
+	      done
+	   fi
+	else
+	   echo -e "${logs_path} is not a file or folder." | tee -a ${clean_up_virt_logs_log}
+	   return 0
+	fi	   
+
+	return ${total_text_file}
+}
+
+#Do logs cleanup on host with regard to /var/lib/libvirt, /var/log/libvirt and /var/log/xen. And calculate the total number of files that are cleaned
+function do_cleanup_on_host() {
+	local extra_logs=($@)
+	local default_logs=("/var/lib/libvirt" "/var/log/libvirt" "/var/log/xen")
+	local ret_result=0
+	local total_cleanup=0
+	local eachlog=""
+
+	for eachlog in ${default_logs[@]};do
+	    echo -e "Going to clean up all emptya and text log files in ${eachlog}." | tee -a ${clean_up_virt_logs_log}
+	    cleanup_logs ${eachlog}
+	    ret_result=$?
+	    echo -e "Cleaned up ${ret_result} empty and text log files in ${eachlog}." | tee -a ${clean_up_virt_logs_log}
+	    total_cleanup=$(( ${total_cleanup} + ${ret_result} ))
+	done
+
+	if [[ ${extra_logs[@]} != "" ]];then
+	   for eachlog in ${extra_logs[@]};do
+	       echo -e "Going to clean up all empty and text log files in ${eachlog}." | tee -a ${clean_up_virt_logs_log}
+	       cleanup_logs ${eachlog}
+	       ret_result=$?
+	       echo -e "Cleaned up $? empty and text log files in ${eachlog}." | tee -a ${clean_up_virt_logs_log}
+	       total_cleanup=$(( ${total_cleanup} + ${ret_result} ))
+	   done
+	fi
```

Comment:
Combine the two arrays ${default_logs[@]} , ${extra_logs[@]} firstly, then use a for loop, similar codes will be saved in above two loops on line 87/96.

Reply:
Combine these two does not produce the expected result, namely reducing many coding lines, the whole extra_logs loop. Because I will add another if..else to judge whether extra should be appended to default logs. Because this functionality is quite simple, it is just another representation of the same problem.

Reply:
I tried your way, it just saved two lines. But anyway I updated it. So done.